Assets/Scripts/Algorithms/Patrolling/Components/PartitionRedistributionComponent.cs

```diff
@@ -0,0 +1,187 @@
+// Copyright 2025 MAEPS
+// 
+// This file is part of MAEPS
+// 
+// MAEPS is free software: you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+// 
+// MAEPS is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License along
+// with MAEPS. If not, see http://www.gnu.org/licenses/.
+//
+// Contributors 2025: 
+// Casper Nyvang Sørensen,
+// Christian Ziegler Sejersen,
+// Jakob Meyer Olsen
+// Mads Beyer Mogensen
+using System.Collections.Generic;
+
+using Maes.Map;
+using Maes.Robot;
+
+using UnityEngine;
+
+
+namespace Maes.Algorithms.Patrolling.Components
+{
+    /// <summary>
+    /// Component responsible for redistributing robots to different partitions.
+    /// </summary>
+    public sealed class PartitionRedistributionComponent : IComponent
+    {
+        // The value is the Id of the Partition and the float is the probability of the robot to redistribute to that partition.
+        private readonly Dictionary<int, float> _redistributionTracker;
+        private readonly IRobotController _controller;
+        private Partition _currentPartition = null!;
+        private readonly List<int> _currentPartitionIntersection;
+        private float _trackerUpdateTimestamp = 0f;
+        private readonly PatrollingMap _map;
+        private Dictionary<int, bool> ReceivedCommunication { get; }
```

Comment:
Why is this a property?

Reply:
It should be a normal field not a property

Reply:
changed it to a field.